rover-apps/common/src/main.cpp

```diff
@@ -0,0 +1,90 @@
+#include "AppConfig.h"
+#include "Logger.h"
+#include "mbed.h"
+
+Thread periodic_10s_thread(osPriorityNormal1);
+Thread periodic_1s_thread(osPriorityNormal2);
+Thread periodic_100ms_thread(osPriorityNormal3);
+Thread periodic_10ms_thread(osPriorityNormal4);
+Thread periodic_1ms_thread(osPriorityNormal5);
+
+void periodic_10s(void) {
+  auto startTime = Kernel::Clock::now();
+  for (Module* module : gModules) {
+    module->periodic_10s();
+  }
+
+  auto nextStartTime = startTime + 10s;
+  if (Kernel::Clock::now() > nextStartTime) {
+    Utility::logger << "Reseting periodic 10s task timing as it failed to hit the deadline!\n";
```

Comment:
unrelated question:
Does our logger support different levels of logging? missed timings is a lot more important than debug type logging.

Reply:
No, it only supports the general printf output

Reply:
Add it to the backlog xd